powersimdata/input/converter/pypsa_to_grid.py

```diff
@@ -230,14 +224,33 @@ def _read_network(self, n, add_pypsa_cols=True):
             bus2sub["sub_id"] = pd.to_numeric(
                 bus2sub.pop("substation").str[3:], errors="ignore"
             )
-            bus2sub_pypsa = bus_pypsa[["substation", "interconnect"]].copy()
-            bus2sub_pypsa["sub_id"] = pd.to_numeric(
-                bus2sub_pypsa.pop("substation").str[3:], errors="ignore"
-            )
         else:
-            warnings.warn("Substations could not be parsed.")
-            sub = pd.DataFrame()
-            bus2sub = pd.DataFrame()
+            # try to parse typical pypsa-eur(-sec) pattern for substations
+            sub_pattern = "[A-Z][A-Z]\d+\s\d+$"
+            interconnect_pattern = "([A-Z][A-Z])"
+
+            sub = bus[bus.index.str.match(sub_pattern)].reindex(columns=sub_cols)
+            sub["interconnect"] = sub.index.str.extract(interconnect_pattern).values
+            sub["name"] = sub.index
+            sub_pypsa = bus_pypsa[bus_pypsa.index.str.match(sub_pattern)][
+                sub_pypsa_cols
+            ]
+
+            sub_pattern = "([A-Z][A-Z]\d+\s\d+)$"
+            bus2sub = pd.DataFrame(
+                {
+                    "name": bus.index.str.extract(sub_pattern)[0].values,
```

Comment:
Should the column name be "sub_id" instead of "name"?

Reply:
Yes, that was the problem thanks!